### ISSUE TITLE
Semicategory op naming

### ISSUE THREAD

**Issue by edmundnoble** [scalaz/scalaz#1946]
What we have right now is really `andThen`, not `compose`, despite being called `compose`. In addition to the naming being confusing, factoring out the `Semicategory` dependency from a method that just uses `Function1` results in the code not working, because `Function1.compose` is the correct order.

**Comment by edmundnoble** [comment]
I propose renaming to the correct order and also introducing an `andThen` operator with flipped order.

**Comment by spf3000** [comment]
@edmundnoble when you say Semicategory - do you mean Semigroup?

**Comment by edmundnoble** [comment]
@spf3000: No, I mean `Semicategory`, see https://github.com/scalaz/scalaz/blob/series/8.0.x/base/shared/src/main/scala/scalaz/tc/semicategory.scala

**Comment by issackurian** [comment]
@edmundnoble Is this still open because I see an `andThen` method on `Semicategory` now.
